### PR TITLE
Upgrade dashboard interaction issue

### LINKED ISSUE #1246: Plotly Dash Versioning Issue
### Overview
Existing plotly dashboards versions are from v1.x while the current plotly dash is on v2.x. There is no longerr any documentation available for older versions of Dash and an upgrade is necessary. 

Currently, I have succesfully upgrade the code and tested in a separate directory with new dashboard incorporated in there. However, then repeating the process in 311 Code base, none of the callback functions get triggered (i.e. no interactivity possible), while the static visuals still get render. I suspect this is something to do with gunicorn interfering with the callbacks being fire, as everything works perfectly without gunicorn. One thing worth mentioning is that this sometimes occur prior to upgrade as well, but the issue will resolve after reloading a few times.

### Action Items
- [ ] Determine which existing dashboards we want to keep
- [x] Upgrade desired dashboards to newest version to work with docker
- [x] Ensure new vesrions of dashboards are compatible with gunicorn 
- [ ] Integrate new dashboards with existing code base

### Resources/Instructions
- See [here](https://github.com/joshuayhwu/311-Data-Dashboard) for Upgraded Dashboards with New Dash Version with Docker 
- https://dash.plotly.com/dash-2-0-migration
- #1378

### PR DESCRIPTION
**Pull request by joshuayhwu**
Fixes #1246 

  - [ ] Resolve callback functions not working with newer versions of dash/gunicorn with flask
  - [ ] All PR Status checks are successful
  - [ ] Peer reviewed and approved

Any questions? See the [getting started guide](https://github.com/hackforla/311-data/blob/master/GETTING_STARTED.md)

### CODE REVIEW

**Comment by joshuayhwu** [comment]
There seems to be some issues with automatic checks on alembic. This might be something to do with database recreation in order to fix the v1 site (See Issue #1239)

**Comment by nichhk** [comment]
Hey I just reran the test, and it passed successfully. I guess the database wasn't initialized properly in your run. The test brings up a local database, so we wouldn't expect any issues with the prod database to affect it.

I'll start reviewing these changes now, thanks!

File: server/dash/app.py
Comment (nichhk) [inline server/dash/app.py:1]: Please alphabetize imports. https://google.github.io/styleguide/pyguide.html#313-imports-formatting

File: server/dash/dashboards/ncSumComp.py
Comment (nichhk) [inline server/dash/dashboards/ncSumComp.py:1]: Can you open a separate PR for adding this file?

File: server/dash/dashboards/neighborhood.py
Comment (nichhk) [inline server/dash/dashboards/neighborhood.py:55]: Too many newlines here.

File: server/dash/dashboards/neighborhood.py
Comment (nichhk) [inline server/dash/dashboards/neighborhood.py:62]: Python uses `snake_case`, so let's use that instead of `camelCase`. i.e., `nc_avg_comp_line_chart`.

File: server/dash/dashboards/neighborhood.py
Comment (nichhk) [inline server/dash/dashboards/neighborhood.py:1]: Ah, I just remembered that Piero is updating this file right now. Let's wait until he merges his PR, and then could you pull and update the dash stuff please?

File: server/dash/index.py
Comment (nichhk) [inline server/dash/index.py:45]: Remove?

File: server/dash/index.py
Comment (nichhk) [inline server/dash/index.py:9]: remove?

File: server/dash/dashboards/neighborhood_recent.py
Comment (nichhk) [inline server/dash/dashboards/neighborhood_recent.py:36]: I know you're just copying this from another file, but could you add a docstring for this?

File: server/dash/dashboards/neighborhood_recent.py
Comment (nichhk) [inline server/dash/dashboards/neighborhood_recent.py:37]: Please capitalize the start of all comments, and add a period at the end.

**Review by nichhk — CHANGES_REQUESTED** [review]
Thanks for figuring this out Josh!

I mentioned this in the comments, but Piero is actively working on updating neighborhoods.py in PR #1242. Let's wait for him to merge that before applying the Dash upgrade to that file.

In general, my comments are about Python style. In particular, we should use `snake_case` for variable names. We should also organize imports as described here:  https://google.github.io/styleguide/pyguide.html#313-imports-formatting.

**Comment by nichhk** [comment]
Could you use an autoformatter for Python in your code editor? I think that will help with some of the style issues like spacing.

File: server/dash/dashboards/neighborhood.py
Comment (joshuayhwu) [reply server/dash/dashboards/neighborhood.py:56]: Thanks for the catch - resolved

File: server/dash/dashboards/neighborhood_recent.py
Comment (joshuayhwu) [reply server/dash/dashboards/neighborhood_recent.py:36]: Thanks for the catch - resolved

File: server/dash/index.py
Comment (joshuayhwu) [reply server/dash/index.py:45]: Resolved - thanks for the catch

**Pull request closed without merging**
